source/extensions/quic_listeners/quiche/envoy_quic_client_session.cc

```diff
@@ -55,9 +55,9 @@ void EnvoyQuicClientSession::OnGoAway(const quic::QuicGoAwayFrame& frame) {
   }
 }
 
-void EnvoyQuicClientSession::OnCryptoHandshakeEvent(CryptoHandshakeEvent event) {
-  quic::QuicSpdyClientSession::OnCryptoHandshakeEvent(event);
-  if (event == HANDSHAKE_CONFIRMED) {
+void EnvoyQuicClientSession::SetDefaultEncryptionLevel(quic::EncryptionLevel level) {
+  quic::QuicSpdyClientSession::SetDefaultEncryptionLevel(level);
+  if (level == quic::ENCRYPTION_FORWARD_SECURE) {
```

Comment:
Are sure this function is only called once with level=FORWARD_SECURE?

The old function name looks like it only happens once; it's not obvious from the new name.

Reply:
Yes, it's only called with FORWARD_SECURE when handshake is finished.

Reply:
SGTM, please add a comment at here and EnvoyQuicServerSession::SetDefaultEncryptionLevel.

Reply:
done